src/main/java/com/google/devtools/build/bfg/JavaSourceFileParser.java

```diff
@@ -108,14 +177,22 @@
       }
       String qualifiedSrc = stripInnerClassFromName(parser.fullyQualifiedClassName);
       dirToClass.put(srcFilePath.getParent(), parser.fullyQualifiedClassName);
+      HashSet<String> adj = new HashSet<>();
+      classToFile.put(qualifiedSrc, srcFilePath.toString());
       for (QualifiedName qualifiedDst : parser.qualifiedTopLevelNames) {
         if (!qualifiedSrc.equals(qualifiedDst.value())) {
-          graph.putEdge(qualifiedSrc, qualifiedDst.value());
+          adj.add(qualifiedDst.value());
         }
       }
+      for (String v : adj) {
+        classToClass.putEdge(qualifiedSrc, v);
+      }
+
```

Comment:
Another nit. 
Can you rename `adj` to `adjacentClasses` and `v` to `qualifiedDst`?

Also, why you can't you add the edge into the graph from within the previous loop? The separate loop seems unnecessary. 

```java
for (QualifiedName qualifiedDst : parser.qualifiedTopLevelNames) {
    if (!qualifiedSrc.equals(qualifiedDst.value())) {
         adjacentClasses.add(qualifiedDst.value());
         classToClass.putEdge(qualifiedSrc, qualifiedDst.value());
    }
}
```

Reply:
> This in turn would lead to test failures as the graph would contain more than it should. Even if isolated nodes are not an immediate problem, it's better not to have them.
> I could have checked beforehand if the node exists in the graph, but at this point having a separate set made sense to me.

I see. That makes sense. 

> Now that I look at it, using a flag makes it look better. Changed, thanks :)

Agreed. I like it.